docs/console-access.md

```diff
@@ -0,0 +1,340 @@
+
+Choice of access control methods. Either QEMU can mandate client certificates
+and perform access control checks on the certificate identity (planned, but not
+yet supported in QEMU & libvirt), or protocol specific authentication must be
+used.
+
+Only supports VNC and SPICE, until QEMU is able to do fine grained ACL checks
+against client certs.
+
+
+### 2. HTTP CONNECT proxy
+
+QEMU on the compute node running a TCP socket with TLS and x509 certificates
+enabled.
+
+Compute node certs must be issued by a CA that is trusted by all tenant
+users and new certs acquired for each new compute node.
+
+One or more proxy servers able to multiplex to an arbitrary number of backend
+compute nodes
+
+Tenant users make HTTP CONNECT request to proxy server, giving the IP address
+and port on the cluster network that QEMU is listening on
+
+Tenant users directly validate the received server certificates against
+the requested IP + port of the QEMU service.
+
+Choice of access control methods. Either QEMU can mandate client certificates
+and perform access control checks on the certificate identity (planned, but not
+yet supported in QEMU & libvirt), or protocol specific authentication must be
+used.
+
+Only supports VNC and SPICE, until QEMU is able to do fine grained ACL checks
+against client certs.
+
+
+### 3. HTTP websockets proxy
+
+QEMU on the compute node running a TCP socket with TLS and x509 certificates
+enabled.
+
+Compute node certs can be issued by a private CA, which only needs to be
+trusted by the proxy servers. Proxies are issued client certs by the CA
+for mutual auth.
+
+One or more proxy servers able to multiplex to an arbitrary number of backend
+compute nodes
+
+Shared secret between the proxy and tenant user identifies the QEMU service to
+connect to and authenticates the user.
+
+Tenant users validate the received server certificates against the IP + port
+of the proxy server.
+
+The proxy server performs TLS handshake with the QEMU servers and validates its
+certificate.
+
+Protocol specific authentication is not exposed to tenant users.
+
+Supports VNC, SPICE and serial consoles with parity of functionality.
+
+
+Proxy Implementation
+--------------------
+
+For both the HTTP CONNECT and HTTP websockets proxy there needs to be some be some
+intelligence in the proxy. In the HTTP CONNECT proxy, there needs to be code to
+validate that the requested IP+port pair actually corresponds to a valid guest
+console connection. This protects against the proxy being used to attack other
+arbitrary network servers running in the cluster network. In the HTTP websockets
+proxy, there needs to be code to validate the shared secret obtained from the
+client, to resolve it into a IP+port for the QEMU connection. It also needs to
+be prepared to do some intelligent MITM of the connections to handle authentication
+and migration.
+
+The libvirt-console-proxy project aims to provide a general purpose server that
```

Comment:
We can use [RBAC](https://kubernetes.io/docs/admin/authorization/rbac/) in combination with a service account on the proxy pods, which can only access the corresponding endpoint in the API server. If such a pod is taken over, it can only get as much information as it could get from a non-compromised resolver. So I don't see much problems with giving it access to the apiserver, I would actually prefer it, to keep things simple. 

/cc @admiyo FYI